Slim/Plugin/AudioScrobbler/Plugin.pm

```diff
@@ -668,6 +709,191 @@ sub _submitNowPlayingError {
 	);
 }
 
+sub submitNowPlayingListenBrainz {
```

Comment:
Instead of having all these `sub doSomethingListenBrainz`, please create a new modules à la `Slim::Plugin::AudioScrobbler::ListenBrainz` for each service or interface in which you have methods of the same name for each service. They you'll only have to define which module to use as `my $serviceModule = 'Slim::Plugin::AudioScrobbler::ListenBrainz;` (or `...::LastFM`, or...), then `$serviceModule->submitNowPlaying()`.

If structured well you could have one implementation (LastFM) be the base class, then have the others inherit from that, only overwriting what needs to be overwritten.

Reply:
I don't know your level of Perl. Please let me know if you want to have some help with this part of it. But you've done great so far 😉.